nengo/transforms.py

```diff
@@ -210,21 +250,12 @@ def sample(self, rng=np.random):
     def toarray(self):
         """Return the dense matrix equivalent of this matrix."""
 
-        dense = np.zeros(self.shape, dtype=self.dtype)
-        dense[self.indices[:, 0], self.indices[:, 1]] = self.data
-        return dense
-
-    @property
-    def dtype(self):
-        return self.data.dtype
+        if self._dense is not None:
+            return self._dense
 
-    @property
-    def ndim(self):
-        return len(self.shape)
-
-    @property
-    def size(self):
-        return self.indices.shape[0]
+        self._dense = np.zeros(self.shape, dtype=self.dtype)
+        self._dense[self.indices[:, 0], self.indices[:, 1]] = self.data
```

Comment:
Should probably mark this as `readonly` as well, since we're caching it.

Reply:
Done